### PR TITLE
Fix hits counter

### PR DESCRIPTION
**Pull request by rdeutz**
# Executive summary

These change allows to count hits for an article also when cache is switched on
# Backwards compatibility

Full b/c, no problems are expected
# Translation impact

none
# Why

If you switch cache on hits for an article are not counted right. The counter will be increased for the first time the article is requested. For the time the article come directly from the cache hits are not counted, this makes the statistic useless.
# Testing instruction
## To reproduce the error
- Install a fresh version of Joomla
- Create an article with an intro and full text, we need a read more button
- Make it a featured article so that you can see the article on the home page
- Notice that the counter is 0 (check this in the backend!)
- Click on the read more button and go back
- Notice that the counter is 1
- Click again and go back
- Counter === 2
- Switch on cache
- Click on the read more button and go back
- Counter === 3
- Click on the read more button and go back
- Counter === 3 (ups)

You can now wait the cache time and click and the counter will increase. If you are not so patient you can clear the cache and click then and the counter will increase
## Expected behavior

Every time you click on an article to counter should increase if cache is set on or off shouldn’t make a difference.
# Apply patch and testing
- Switch cache off
- check the hit counter value => N
- Click on the read more button and go back
- Counter === N+1
- Switch on cache
- Click on the read more button and go back
- Counter === N+2
- Click on the read more button and go back
- Counter === N+3

Bingo :-)

All counter checks should be made in the backend the frontend is caching values if caching is on.

### CODE REVIEW

File: components/com_content/controller.php [hunk 1]
```diff
@@ -101,6 +101,15 @@ public function display($cachable = false, $urlparams = false)
 			return JError::raiseError(403, JText::sprintf('JLIB_APPLICATION_ERROR_UNHELD_ID', $id));
 		}
 
+		if ($vName == 'article' && $cachable && true)
```
Comment (Fedik) [inline components/com_content/controller.php]: `true` or not `true` :wink:

Comment (rdeutz) [inline components/com_content/controller.php]: oh shit, added to make it false or true for testing :-) Thanks for spotting it

Comment (andrepereiradasilva) [inline components/com_content/controller.php]: :smile:

**Comment by infograf768** [comment]
I have tested this item :red_circle: unsuccessfully on f8808d621d73931cf2f9483c923049124b026194<br /><br />This did not work here.
I did not use the Read More, just re-loaded the article.
Before implementing Conservative Cache, all is fine. Hits are updated each time I load the article.
After, it is not updated, patched or not.
Hits are incremented OK in back-end.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8887">issues.joomla.org/joomla-cms/8887</a>.</sub>

**Comment by rdeutz** [comment]
@infograf768 and how did you load the article, when not with clicking the read more button?

**Comment by infograf768** [comment]
Just reloading the page. (cmd-R on Firefox Macintosh)

**Comment by Bakual** [comment]
@infograf768 I'd say it's expected that the hit counter that is displayed in frontend is also cached and stays on the old value. If it does a jump to the correct value after cache is expired.then it's fine.

**Comment by infograf768** [comment]
@Bakual
If it is expected, then I do not see the value of this PR... See Tests instructions above.

**Comment by rdeutz** [comment]
The value of the PR is that it counts also when the cache is on, you need to check the count in the backend

**Comment by infograf768** [comment]
In that case, it is indeed working.
Please change test instructions.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on f8808d621d73931cf2f9483c923049124b026194<br /><br />Hots counter now works in back-end.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8887">issues.joomla.org/joomla-cms/8887</a>.</sub>

**Comment by rdeutz** [comment]
Changed Test instructions, thanks for testing @infograf768

**Comment by justinherrin** [comment]
I have tested this item :white_check_mark: successfully on f8808d621d73931cf2f9483c923049124b026194<br /><br />Hit counter in backend increases properly now even if cache is turned on. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8887">issues.joomla.org/joomla-cms/8887</a>.</sub>

**Comment by brianteeman** [comment]
RTC - thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8887">issues.joomla.org/joomla-cms/8887</a>.</sub>

**Comment by infograf768** [comment]
Milestone to bet set to 3.5.0 as this is a bug correction.

**Pull request merged**